client/pages/restaurant-page.jsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import Container from 'react-bootstrap/Container';
+import fetch from 'node-fetch';
+import Spinner from './spinner';
+import Row from 'react-bootstrap/Row';
+import Column from 'react-bootstrap/Col';
+
+export default class Restaurant extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedRestaurant: null
+    };
+  }
+
+  componentDidMount() {
+    fetch('/api/random')
+      .then(response => response.json())
+      .then(data => this.setState({
+        selectedRestaurant: data
+      }));
+  }
+
+  render() {
+    const { selectedRestaurant } = this.state;
+    const categories = [];
+
+    function tConv24(time24) {
+      let ts = time24;
+      const H = +ts.substr(0, 2);
+      let h = (H % 12) || 12;
+      h = (h < 10) ? ('0' + h) : h;
+      const ampm = H < 12 ? ' AM' : ' PM';
+      ts = h + ':' + ts.substr(2, 3) + ampm;
+      return ts;
+    }
+
+    if (!selectedRestaurant) {
+      return (
+      <Spinner/>
+      );
+    }
+    const zipIndex = selectedRestaurant.location.display_address.length - 1;
+
+    const days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
+
+    const hoursByDay = days.map((day, dayIndex) => {
+      const hours = selectedRestaurant.hours.open.find(timeSlot => timeSlot.day === dayIndex);
+      return {
+        day,
+        hours
+      };
+    });
+
+    for (let i = 0; i < selectedRestaurant.categories.length; i++) {
+      categories.push(selectedRestaurant.categories[i].title);
+    }
```

Comment:
The `categories` variable is waaaaayyy up at the top of the render method. How about just using `.map()` right here to create that variable. Loop-and-push is basically what it's for.